config/sync/views.view.uploaded_products.yml

```diff
@@ -311,11 +363,17 @@ display:
           relationship: none
           group_type: group
           admin_label: ''
-          empty: false
+          empty: true
           tokenize: false
           content:
             value: ''
             format: full_html
+          roles_fieldset:
+            roles:
+              administrator: administrator
+              product_uploader: product_uploader
```

Comment:
Also please make sure that users that should not see the button, can't even access the product creation page with its URL.

Reply:
I think it works fine: if a normal user tries to access the product creation page, (s)he gets the "Access denied You are not authorized to access this page." message.